src/helpers.js

```diff
@@ -34,15 +34,14 @@ export const cloneTr = tr => {
 // (position: number, content: union<ProseMirrorNode, Fragment>) → (tr: Transaction) → Transaction
 // Returns a `replace` transaction that replaces a node at a given position with the given `content`.
 // It will return the original transaction if replacing is not possible.
-// `position` should point at the start of a node in the document.
+// `position` should point at the position immediately before the node.
 export const replaceNodeAtPos = (position, content) => tr => {
-  const before = position - 1;
-  const node = tr.doc.nodeAt(before);
-  const $pos = tr.doc.resolve(before);
+  const node = tr.doc.nodeAt(position);
+  const $pos = tr.doc.resolve(position);
   if (canReplace($pos, content)) {
-    tr = tr.replaceWith(before, before + node.nodeSize, content);
-    const start = tr.selection.$from.pos - 1;
-    return cloneTr(setTextSelection(Math.max(start, 0), -1)(tr));
+    tr = tr.replaceWith(position, position + node.nodeSize, content);
+    const pos = tr.selection.$from.pos - 1;
+    return cloneTr(setTextSelection(Math.max(pos, 0), -1)(tr));
```

Comment:
what selection position are we adjusting here? Why would we change the TextSelection when it's say, at the end of a document, if we replace a node at the start?

Are you wanting to handle the movement of the selection after the node has been replaced?

Then you'll probably want
```
const cursor = tr.mapping.map(tr.selection.from, -1);
return cloneTr(setTextSelection(cursor)(tr));
```

Reply:
fair enough, I'll update that